packages/storefront-sdk/src/types/after.ts

```diff
@@ -0,0 +1,37 @@
+import type {
+	SpaceProperties,
+	Product,
+	Content,
+	ContentEdge,
+	NavigationGroup,
+	ProductEdge,
+	ProductCollection,
+	ProductCollectionEdge
+} from './storefront.js';
+import type { dataFetchingMethods } from '../utils/index.js';
+
+export type DataFetchingMethodName = typeof dataFetchingMethods[number];
+
+export type AfterCallback<T> = (responseObj: T) => Promise<T> | T;
+
+export type MethodData = {
+	products: Product[] | ProductEdge[];
+	productCollections: ProductCollection[] | ProductCollectionEdge[];
+	productCollectionEntries: Product[] | ProductEdge[];
+	content: Content[] | ContentEdge[];
+	navigation: NavigationGroup[];
+	spaceProperties: SpaceProperties;
+	query: Record<string, unknown> | Array<Record<string, unknown>>;
+};
```

Comment:
We probably will need to rethink this once we have a clearer picture of how plugins will work, and whether these methods (other than `.query`) will be made available by a plugin.